### PR TITLE
Pass results array to Eagerloader instead of statement.

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
@othercorey Sorry if you were already working on this but I was just giving it a try and it turned out lot easier than I had thought.

I'll finish the docblock and tests updates if this seems fine in general.

**Comment by othercorey** [comment]
> @othercorey Sorry if you were already working on this but I was just giving it a try and it turned out lot easier than I had thought.

I'm not going to work on Statement until well after you're done.

File: src/Database/Statement/SqliteStatement.php [hunk 1]
```diff
@@ -28,18 +28,18 @@ class SqliteStatement extends StatementDecorator
     /**
      * @inheritDoc
      */
-    public function execute(?array $params = null): bool
-    {
-        if ($this->_statement instanceof BufferedStatement) {
-            $this->_statement = $this->_statement->getInnerStatement();
-        }
+    // public function execute(?array $params = null): bool
```
Comment (markstory) [inline src/Database/Statement/SqliteStatement.php]: This looks like it could be cleaned up.

**Review by markstory — COMMENTED** [review]
Looking good to me :clap:

File: src/Database/Statement/SqliteStatement.php (hunk 1 above)
Comment (ADmad) [reply src/Database/Statement/SqliteStatement.php]: Yeah I'll deal with this in a separate PR.

File: src/ORM/Query.php [hunk 2]
```diff
@@ -1131,8 +1132,13 @@ protected function _execute(): ResultSetInterface
             return new $decorator($this->_results);
         }
 
-        $statement = $this->getEagerLoader()->loadExternal($this, $this->execute());
-        $resultset = new ResultSet($this, $statement->fetchAll('assoc') ?: []);
+        $results = $this->execute()->fetchAll(StatementInterface::FETCH_TYPE_ASSOC);
+        if ($results === false) {
```
Comment (saeideng) [inline src/ORM/Query.php:1136]: why checking for `false` ?

Comment (othercorey) [reply src/ORM/Query.php:1136]: php8 changed the return type of fetchAll() to always be an array. Don't know if you made this change already.

Comment (ADmad) [reply src/ORM/Query.php:1136]: The return type of `StatementInterface::fetchAll()` is still `array|false`. So the check for `false` is needed for now to keep static analyzers happy.

We can change that in a separate PR.

**Comment by othercorey** [comment]
This looks like what I wanted to do with EagerLoader.

**Pull request merged**